pkg/sanity/controller.go

```diff
@@ -210,6 +222,7 @@ var _ = DescribeSanity("Controller Service", func(sc *SanityContext) {
 				},
 			)
 			Expect(err).NotTo(HaveOccurred())
+			cl.RegisterVolume(name, VolumeInfo{VolumeID: vol.GetVolume().GetId()})
 			Expect(vol).NotTo(BeNil())
```

Comment:
All of the test cases are validating these same things. Can the validation be put into a common function?

Reply:
I was wondering the same about the full test and ended up moving that to a single test and then again here after making the same change over and over again.

I guess some more tests could be eliminated, but I haven't looked too closely and would prefer to get this PR merged first before doing further simplifications.

Reply:
I don't mean eliminating tests, but consolidating the 4 checks about create volume return values into a common function

Reply:
The downside of putting the checks into a common function is that line information in case of a failure isn't useful anymore:
```
• Failure [0.017 seconds]
Controller Service
/fast/work/gopath/src/github.com/intel/open-infrastructure-offload/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/tests.go:44
  CreateVolume
  /fast/work/gopath/src/github.com/intel/open-infrastructure-offload/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/controller.go:173
    should return appropriate values SingleNodeWriter WithCapacity 1Gi Type:Mount [It]
    /fast/work/gopath/src/github.com/intel/open-infrastructure-offload/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/controller.go:250

    Expected an error to have occurred.  Got:
        <nil>: nil

    /fast/work/gopath/src/github.com/intel/open-infrastructure-offload/vendor/github.com/kubernetes-csi/csi-test/pkg/sanity/controller.go:73
```

This line 73 is in a new helper function where I intentionally inverted the error check to trigger a test failure.

I think this needs further discussion, so I'll leave it as it is.